src/lizaalert/users/views.py

```diff
@@ -152,3 +154,40 @@ def get_queryset(self):
             volunteer_badge = volunteer_badge.filter(course_id=course_id)
         queryset = super().get_queryset().filter(id__in=volunteer_badge.values("badge_id"))
         return queryset
+
+
+class BadgeVolunteerListView(ListAPIView):
+    """
+    Отображение списка волонтёров по ачивке (slug-полю поиска).
+
+    Методы:
+    - GET: Возвращает список волонтёров.
+
+    Параметры пагинации:
+    - limit: количество объектов на странице,
+    - offset: отступ первой страницы.
+
+    Пример: /?limit=3&offset=2
+
+    """
+
+    serializer_class = VolunteerSerializer
+    pagination_class = LimitOffsetPagination
+
+    @swagger_auto_schema(
+        operation_description="""
+        Получение QuerySet для волонтеров имеющих определенный бэйдж.
+
+        Возвращает:
+            QuerySet: QuerySet для волонтеров отфильтрованных по полю "badge_slug" в модели Badge.
+        """,
+        responses={
+            200: VolunteerSerializer(),
+            400: "Bad Request",
+        },
+    )
+    def get_queryset(self):
+        queryset = Volunteer.objects.all()
+        badge_slug = self.kwargs["badge_slug"]
+        queryset = queryset.filter(badges__badge_slug=badge_slug)
```

Comment:
здесь лучше объединить строкии

```suggestion
        badge_slug = self.kwargs["badge_slug"]
        queryset = Volunteer.objects.filter(badges__badge_slug=badge_slug)
```